src/compute/src/sink/correction.rs

```diff
@@ -263,9 +270,14 @@ impl<D: Ord> ConsolidatingVec<D> {
     }
 
     /// Return an iterator over the borrowed items.
-    fn iter(&self) -> impl Iterator<Item = &(D, Diff)> {
+    pub fn iter(&self) -> impl Iterator<Item = &(D, Diff)> {
         self.0.iter()
     }
+
+    /// Returns mutable access to the underlying items.
+    pub fn as_mut(&mut self) -> &mut Vec<(D, Diff)> {
+        &mut self.0
+    }
```

Comment:
Could this be an `iter_mut` instead? For the CT usage that seems sufficient, and it would ensure that the caller can't violate any assumptions about the `Vec` we might have.

Reply:
Oh yeah, I like that better. I semi-convinced myself that nothing bad can happen by the caller of `as_mut` changing the len/cap on us, but the only thing that this can do is make is less consolidated, which is definitely fine.